core/src/main/java/jenkins/websocket/WebSocketSession.java

```diff
@@ -114,45 +115,45 @@ protected void error(Throwable cause) {
         LOGGER.log(Level.WARNING, "unhandled WebSocket service error", cause);
     }
 
-    protected void binary(byte[] payload, int offset, int len) {
+    protected void binary(byte[] payload, int offset, int len) throws IOException {
         LOGGER.warning("unexpected binary frame");
     }
 
-    protected void text(String message) {
+    protected void text(String message) throws IOException {
         LOGGER.warning("unexpected text frame");
     }
 
     @SuppressWarnings("unchecked")
-    protected final Future<Void> sendBinary(ByteBuffer data) {
+    protected final Future<Void> sendBinary(ByteBuffer data) throws IOException {
         try {
             return (Future<Void>) remoteEndpoint.getClass().getMethod("sendBytesByFuture", ByteBuffer.class).invoke(remoteEndpoint, data);
         } catch (Exception x) {
-            throw new RuntimeException(x);
+            throw new IOException(x);
         }
     }
 
-    protected final void sendBinary(ByteBuffer partialByte, boolean isLast) {
+    protected final void sendBinary(ByteBuffer partialByte, boolean isLast) throws IOException {
         try {
             remoteEndpoint.getClass().getMethod("sendPartialBytes", ByteBuffer.class, boolean.class).invoke(remoteEndpoint, partialByte, isLast);
         } catch (Exception x) {
-            throw new RuntimeException(x);
+            throw new IOException(x);
```

Comment:
Cleaner generally, though I would still recommend patching `SlaveComputer` to catch any `Exception`, just in case.

Reply:
Done